src/settingsFields.component.js

```diff
@@ -297,15 +297,24 @@ class SettingsFields extends React.Component {
                 // Base config, common for all component types
                 const validators = []
                 if (mapping.validators) {
-                    mapping.validators.forEach((name) => {
-                        if (wordToValidatorMap.has(name)) {
-                            const validator = wordToValidatorMap.get(name)
+                    mapping.validators.forEach((validator) => {
+                        if (
+                            typeof validator === 'string' &&
+                            wordToValidatorMap.has(validator)
+                        ) {
+                            const d2Validator =
+                                wordToValidatorMap.get(validator)
                             validators.push({
-                                validator,
+                                validator: d2Validator,
                                 message: translateValidatorMessage(
-                                    validator.message
+                                    d2Validator.message
                                 ),
                             })
+                        } else if (typeof validator === 'function') {
+                            validators.push({
+                                validator,
+                                message: validator.message,
+                            })
                         }
                     })
```

Comment:
This change was slightly unfortunate. It turned out this app was solely relying on the deprecated d2 validators, so I had to invent a small escape hatch to use local validators.